Specifications/model/information.model.md

```diff
@@ -0,0 +1,151 @@
+  a dataspace. The identity standard used by a provider is not defined but could be _OAuth2_ or _Decentralized Identifiers using did:web_. An identity provider may be a third-party
+  or a participant itself (for example, in the case of decentralized identifiers).
+- A `Credential Issuer` issues _verifiable credentials_ used by participant agents to allow access to assets and verify usage control.
+
+The diagram below depicts the relationships between `ParticipantAgent` types:
+
+![](./im.participant.entities.png)
+
+- A `CatalogService` is a `Participant Agent` that makes a [DCAT Catalog](https://www.w3.org/TR/vocab-dcat-3/#Class:Catalog) available to other participants.
+- A `Catalog` contains one or more `Asset Entries`, which are [DCAT Datasets](https://www.w3.org/TR/vocab-dcat-3/#Class:Dataset). A `Catalog` also contains **_at least one_**
+  [DCAT DataService](https://www.w3.org/TR/vocab-dcat-3/#Class:Data_Service) that references a `Connector` where assets may be obtained.
+- An `Asset Entry` has **one** `Offer`, which is an [ODRL Offer](https://www.w3.org/TR/odrl-model/#policy-offer) describing the _usage control policy_ associated with the asset.
+- A `Connector` is a `Participant Agent` that performs `Contract Negotiation` and `Asset Transfer` operations with another connector. An outcome of a `ContractNegotiation` may
+  be the production of an `Agreement`, which is an [ODRL Agreement](https://www.w3.org/TR/odrl-model/#policy-agreement) defining the _usage control policy_ agreed to for an asset.
+
+## 2.2 Classes
+
+Not all dataspace entities have a concrete _technical_ materialization; some entities may exist as purely logical constructs. For example, a `Dataspace Authority`
+and `Participant Agent` have no representation in the protocol message flows that constitute dataspace interactions. This section outlines the classes that comprise the concrete
+elements of the information model, i.e. those that are represented in protocol message flows.
+
+### 2.2.1 Catalog
+
+A `Catalog` is a [DCAT Catalog](https://www.w3.org/TR/vocab-dcat-3/#Class:Catalog) with the following attributes:
+
+- 0..N  `Asset Entries`. Since a catalog may be dynamically generated for a request based on the requesting participant's credentials it is possible for it to contain 0 matching
+  asset entries.  (DCAT PROFILE)
+- 1.N [DCAT DataService](https://www.w3.org/TR/vocab-dcat-3/#Class:Data_Service) that references a `Connector` where assets may be obtained.  (DCAT PROFILE)
+
+### 2.2.1 Asset Entry
+
+An `Asset Entry` is a [DCAT Dataset](https://www.w3.org/TR/vocab-dcat-3/#Class:Dataset) with the following attributes:
+
+- 1..N `hasPolicy` attributes that contain an ODRL `Offer` defining the usage control policy associated with the asset. **_Offers must NOT contain any target attributes. The
+  target of an offer is the asset associated with the containing asset entry._** (ODRL PROFILE)
```

Comment:
This means that an Offer can only have one asset but there is a need to maintain one offer for a set of (small/cheap) data assets.

Reply:
You could either negotiate the usage of each asset in a separate negotiation process. Or you define a set of assets as one asset, to again have the 1..1 relation.

Reply:
Why not allowing more than one entry? Either a set with explicit mentions of the data asset ids or a pattern (all data assets which ids follow the pattern 'data_asset_[0-9]')?

Reply:
We should be careful with definitions and their relations. As @juliapampus alluded to, an asset (dataset) is the atomic unit of data sharing. An offer is only associated with one asset. An asset may be associated (hasPolicy) with a multiplicity of offers. By extension, a negotiation involves **exactly one** asset. That said, an asset may consist of a single data file, a collection of files, or a non-finite data stream. To keep things manageable from both a specification and implementation perspective, asset "atomicity" must be a conceptual bedrock. Beyond this, the specifications do not need to model the contents of an asset.      

Solving the (important) use case where someone wants to have an easy way to define a policy that is "applied" to a group of assets is an implementation detail that I do not think should be defined in the specification as it is a management feature. 

In the EDC, we enabled this use case by introducing the concept of a [ContractDefinition](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/a418469c8929403613bce132e8fa55678e24a2cf/spi/control-plane/contract-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/domain/contract/offer/ContractDefinition.java). A `ContractDefinition` is associated with an access policy, a contract policy, and an asset selector expression. The latter is used to resolve a set of assets the definition applies to (basically a query). `Offers` are generated from that combination on the fly.

Other implementations may choose a different approach. We must associate an `Offer` with exactly one asset. How an `Offer` is created is an implementation detail we don't need to address in the specification.